Targets/QJS/Patches/Fuzzilli-instrumentation-for-QJS.patch

```diff
@@ -131,198 +144,192 @@ index 7e4839f..7dbf2b6 100644
-+        } else {
-+          fprintf(stderr, "Failed to eval_buf reprl\n");
++            if (expr) {
++                if (eval_buf(ctx, expr, strlen(expr), "<cmdline>", 0))
++                    goto fail;
++            } else
++            if (optind >= argc) {
++                /* interactive mode */
++                interactive = 1;
++            } else {
++                const char *filename;
++                filename = argv[optind];
++                if (eval_file(ctx, filename, module))
++                    goto fail;
++            }
++            // if (interactive) {
++            //     js_std_eval_binary(ctx, qjsc_repl, qjsc_repl_size, 0);
++            // }
++            js_std_loop(ctx);
          }
 -        js_std_loop(ctx);
-+        js_free(ctx, script_src);
-+        fflush(stdout);
-+        fflush(stderr);
-+        // Send return code to parent and reset edge counters.
-+        status = (result & 0xff) << 8;
-+        CHECK(write(REPRL_CWFD, &status, 4) == 4);
-+        __sanitizer_cov_reset_edgeguards();
-+      } while (reprl_mode);
-+    } else {
-+      if (expr) {
-+        if (eval_buf(ctx, expr, strlen(expr), "<cmdline>", 0)) goto fail;
-+      } else if (optind >= argc) {
-+        /* interactive mode */
-+        interactive = 1;
-+        // Google Change
-+        // if (interactive) {
-+        //  js_std_eval_binary(ctx, qjsc_repl, qjsc_repl_size, 0);
-+        // }
-+        // End Google Change
-+      } else {
-+        const char *filename;
-+        filename = argv[optind];
-+        if (eval_file(ctx, filename, module)) goto fail;
-+      }
-+      js_std_loop(ctx);
      }
-+  }
      
      if (dump_memory) {
-         JSMemoryUsage stats;
 diff --git a/quickjs.c b/quickjs.c
-index 8fbb7a3..a9d1110 100644
+index 7916013..f84825a 100644
 --- a/quickjs.c
 +++ b/quickjs.c
-@@ -22,11 +22,13 @@
-  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
-  * THE SOFTWARE.
-  */
-+
- #include <stdlib.h>
- #include <stdio.h>
+@@ -27,6 +27,7 @@
  #include <stdarg.h>
  #include <inttypes.h>
  #include <string.h>
 +#undef NDEBUG
  #include <assert.h>
  #include <sys/time.h>
  #include <time.h>
-@@ -34815,6 +34817,51 @@ static JSValue JS_ReadFunctionTag(BCReaderState *s)
-     JS_FreeValue(ctx, obj);
+@@ -35408,6 +35409,52 @@ static JSValue JS_ReadFunctionTag(BCReaderState *s)
      return JS_EXCEPTION;
  }
+ 
 +#define REPRL_DWFD 103
 +static JSValue js_fuzzilli(JSContext *ctx, JSValueConst this_val, int argc,
 +                           JSValueConst *argv) {
-+  printf("js_fuzzilli\n");
-+  const char* str = JS_ToCString(ctx, argv[0]);
-+  if (!str) {
-+    printf("js_fuzzilli NO CMD\n");
-+    return JS_FALSE;
-+  }
-+  if (!strcmp(str, "FUZZILLI_CRASH")) {
-+    printf("js_fuzzilli CRASH\n");
-+    int type;
-+    if (JS_ToInt32(ctx, &type, argv[1])) {
-+      JS_FreeCString(ctx, str);
-+      printf("js_fuzzilli wrong type\n");
-+      return JS_FALSE;
++    printf("js_fuzzilli\n");
```

Comment:
Nit: should we drop these printfs? The crash testers in https://github.com/googleprojectzero/fuzzilli/blob/8431a409dc431b04438511a684af598428aa147f/Sources/FuzzilliCli/Profiles/QjsProfile.swift#L34 should notice if this logic breaks I think

Reply:
Yep, these were in the original patch as well.  I've removed them.